models/user/user.go

```diff
@@ -1219,6 +1227,43 @@ func GetAdminUser() (*User, error) {
 	return &admin, nil
 }
 
+func isUserVisibleToViewerCond(viewer *User) builder.Cond {
+	cond := builder.NewCond()
+	if viewer != nil && viewer.IsAdmin {
+		return cond
+	}
+	cond = builder.Eq{
+		"`user`.Visibility": structs.VisibleTypePublic,
+	}
+
+	if viewer == nil || viewer.IsRestricted {
+		return cond
+	}
+
+	cond = builder.Not{builder.Eq{
+		"`user`.Visibility": structs.VisibleTypePrivate,
+	}}.Or(
```

Comment:
Please use `.And(builder.Or(` otherwise the previous statement won't be considered for all these users.

Reply:
This is deliberate.

Reply:
The previous statement is deliberately not being considered.

Reply:
Saw the code wrong.

Reply:
No worries I had it in a confusing way and have simplified it now

Reply:
Just to add, somehow it's working. But looking at the code it almost seems like that every public/limited team/org user will get included into the follow{ing,ers}, given the `.Or(...)`